argilla/src/argilla/records/_io/_datasets.py

```diff
@@ -64,20 +182,18 @@ def _record_dicts_from_datasets(dataset: HFDataset) -> List[Dict[str, Union[str,
         Returns:
             Generator[Dict[str, Union[str, float, int, list]], None, None]: A generator of dictionaries to be passed to DatasetRecords.add or DatasetRecords.update.
         """
-        media_features = HFDatasetsIO._get_image_features(dataset)
-        if media_features:
-            dataset = HFDatasetsIO._cast_images_as_urls(hf_dataset=dataset, columns=media_features)
+        hf_dataset = HFDatasetsIO.to_argilla(hf_dataset=hf_dataset)
         try:
-            dataset: IterableDataset = dataset.to_iterable_dataset()
+            hf_dataset: IterableDataset = hf_dataset.to_iterable_dataset()
         except AttributeError:
             pass
         record_dicts = []
-        for example in dataset:
+        for example in hf_dataset:
             record_dicts.append(example)
```

Comment:
```suggestion
        record_dicts.append([example for example in hf_dataset])
```

Reply:
Dataset samples are serialised on iteration. I can use a comprehension if that prettier.